### PR TITLE
stm32 xspi reference  to replace the ospi for stm32x mcu

### PR DESCRIPTION
**Pull request by FRASTM**
This change is for preparing the future stm32 series that could have spi/quad/octo/hexa spi peripheral and gives more flexibility to stm32 devices

The OSPI_xxx constant are renamed with XSPI_ prefix

The XSPI constants are common values to the existing zephyr  flash_stm32_qspi or flash_stm32_ospi.

requires https://github.com/zephyrproject-rtos/zephyr/pull/70434

### CODE REVIEW

**Comment by FRASTM** [comment]
rebased on aeea1d10de8859d9368cbf07be0b9c3726c79a67

**Comment by FRASTM** [comment]
CI failure due to https://github.com/zephyrproject-rtos/zephyr/issues/70445

File: include/zephyr/dt-bindings/flash_controller/ospi.h
Comment (erwango) [inline include/zephyr/dt-bindings/flash_controller/ospi.h:1]: Don't remove purely, but add a deprecation notice + update migration guide

File: dts/arm/st/h5/stm32h562.dtsi [hunk 1]
```diff
@@ -5,7 +5,7 @@
  */
 
 #include <st/h5/stm32h5.dtsi>
-#include <zephyr/dt-bindings/flash_controller/ospi.h>
+#include <zephyr/dt-bindings/flash_controller/xspi.h>
```
Comment (erwango) [inline dts/arm/st/h5/stm32h562.dtsi:8]: Keep both during the deprecation period ?

**Review by erwango — CHANGES_REQUESTED** [review]

File: include/zephyr/dt-bindings/flash_controller/ospi.h
Comment (FRASTM) [reply include/zephyr/dt-bindings/flash_controller/ospi.h:1]: updating the doc/releases/migration-guide-3.7 to introduce the new XSPI__ prefix

**Comment by FRASTM** [comment]
Closing this PR and change to XSPI for the stm32h5 serie only : see PR #70371

**Pull request closed without merging**